internal/controller/connector_controller.go

```diff
@@ -0,0 +1,299 @@
+	k8sErrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+
+	appsv1beta1 "github.com/hstreamdb/hstream-operator/api/v1beta1"
+)
+
+// ConnectorReconciler reconciles a Connector object
+type ConnectorReconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
+}
+
+//+kubebuilder:rbac:groups=apps.hstream.io,resources=connectors,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=apps.hstream.io,resources=connectors/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=apps.hstream.io,resources=connectors/finalizers,verbs=update
+//+kubebuilder:rbac:groups="",resources=configmaps,verbs=create;update;patch;delete
+//+kubebuilder:rbac:groups="",resources=deployments,verbs=create;update;patch;delete
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+// TODO(user): Modify the Reconcile function to compare the state specified by
+// the Connector object against the actual cluster state, and then
+// perform operations to make the cluster state reflect the state specified by
+// the user.
+//
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.16.0/pkg/reconcile
+func (r *ConnectorReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	log = logf.FromContext(ctx)
+
+	var connector appsv1beta1.Connector
+	if err := r.Get(ctx, req.NamespacedName, &connector); err != nil {
+		if !k8sErrors.IsNotFound(err) {
+			log.Error(err, "fail to fetch Connector")
+		}
+
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	if !connector.ObjectMeta.DeletionTimestamp.IsZero() {
+		return ctrl.Result{}, nil
+	}
+
+	configMapNames := []string{}
+	for _, stream := range connector.Spec.Streams {
+		configMapNames = append(configMapNames, appsv1beta1.GenConnectorConfigMapName(connector.Spec.TemplateName, false)+"-for-"+stream)
+	}
+	var configs []map[string]interface{}
+	if connector.ObjectMeta.DeletionTimestamp.IsZero() {
+		cfgs, err := r.mergePatchesIntoConfigs(ctx, connector)
+		if err != nil {
+			log.Error(err, "fail to merge patches into config")
+
+			return ctrl.Result{}, err
+		}
+		configs = cfgs
+	}
+
+	for index, configMapName := range configMapNames {
+		var connectorConfigMap corev1.ConfigMap
+		if err := r.Get(ctx, types.NamespacedName{
+			Namespace: connector.Namespace,
+			Name:      configMapName,
+		}, &connectorConfigMap); err != nil {
+			if !k8sErrors.IsNotFound(err) {
+				log.Error(err, "fail to fetch ConfigMap which generated by Connector",
+					"Connector", connector.Name,
+					"ConfigMap", configMapName,
+				)
```

Comment:
If enable Debug log, the error will print, so we don't need print log in our code